### PR TITLE
Thorough analyzed dependency checks

### PR DESCRIPTION
**Pull request by tisnik**
More thorough test of analyzed and unknown dependencies + new test step to check unknown dependencies.

### CODE REVIEW

**Comment by tisnik** [comment]
[test]

**Comment by tisnik** [comment]
[test]

File: integration-tests/features/steps/stack_analysis.py [hunk 1]
```diff
@@ -650,6 +650,19 @@ def check_analyzed_dependencies_count(context, expected=1):
             analyzed_dependencies_count, expected)
 
 
+@then("I should find exactly one unknown dependency")
```
Comment (miteshvp) [inline integration-tests/features/steps/stack_analysis.py]: You can never be sure an unknown will remain unknown for a lifetime.

File: integration-tests/features/steps/stack_analysis.py [hunk 2]
```diff
@@ -650,6 +650,19 @@ def check_analyzed_dependencies_count(context, expected=1):
...
+@then("I should find exactly {expected:n} unknown dependencies")
+def check_unknown_dependencies_count_exact_check(context, expected=1):
+    """Check number of unknown dependencies."""
+    jsondata = context.response.json()
+    assert jsondata is not None
+    path = "result/0/user_stack_info/unkwnown_dependencies_count"
+    unknown_dependencies_count = get_value_using_path(jsondata, path)
+    assert unknown_dependencies_count == expected, \
```
Comment (miteshvp) [inline integration-tests/features/steps/stack_analysis.py]: Rather the test should be `unknown_dependencies_count <= expected`

File: integration-tests/features/steps/stack_analysis.py (hunk 1 above)
Comment (tisnik) [reply integration-tests/features/steps/stack_analysis.py]: We already have the following test step:
`Then I should find no more than X unknown dependencies`

The one with exact one unknown dependency is to check the following file https://github.com/fabric8-analytics/fabric8-analytics-common/pull/988/files#diff-c1053ef11641602c7a94f1ba11fc57c3 ie the one where component is non existent (we can use even more unlike name of course :)

File: integration-tests/features/steps/stack_analysis.py (hunk 2 above)
Comment (tisnik) [reply integration-tests/features/steps/stack_analysis.py]: it is defined below:

`@then("I should find no more than {expected:n} unknown dependencies")`

File: integration-tests/features/steps/stack_analysis.py (hunk 1 above)
Comment (miteshvp) [reply integration-tests/features/steps/stack_analysis.py]: Ok, in that case we could rename this to something like
`@then("I should find exactly one really unknown dependency")`

File: integration-tests/features/steps/stack_analysis.py (hunk 2 above)
Comment (miteshvp) [reply integration-tests/features/steps/stack_analysis.py]: Ok

File: integration-tests/features/steps/stack_analysis.py (hunk 1 above)
Comment (tisnik) [reply integration-tests/features/steps/stack_analysis.py]: Ok I'm updating the PR. Thanks @miteshvp

**Comment by tisnik** [comment]
[test]

**Pull request merged**